extensions/amp-form/0.1/validation-bubble.js

```diff
@@ -51,10 +51,10 @@ export class ValidationBubble {
     this.isVisible_ = false;
 
     /** @private @const {!Element} */
-    this.bubbleElement_ = win.document.createElement('div');
+    this.bubbleElement_ = ampdoc.win.document.createElement('div');
     this.bubbleElement_.classList.add('-amp-validation-bubble');
     this.bubbleElement_[OBJ_PROP] = this;
-    win.document.body.appendChild(this.bubbleElement_);
+    ampdoc.win.document.body.appendChild(this.bubbleElement_);
```

Comment:
No, after checking I found that the bubbles need to be in the `ampdoc.getBody()`, because the bubble DOM node needs to be inside the shadow root to be styled properly.

I also found `viewport.getLayoutRect` returns the position relative to the shadow root, so the bubbles are positioned with an incorrect offset.